network_analysis/nbconverted/1_preprocess_network_data.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Preprocess and validate eADAGE network data
+# 
+# eADAGE G-G network/edge list was generated by Alex using the script at `../generic_expression_patterns_modules/make_GiG_network.R`. 
+# 
+# Here, we preprocess it to reweight negative edges, add generic gene info from `../pseudomonas_analysis/annot_df.tsv`, and save in a file that can be quickly loaded by `graph-tool` for subsequent analyses.
+
+# In[1]:
+
+
+import os
+
+import numpy as np
+import pandas as pd
+import graph_tool.all as gt
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+# In[2]:
+
+
+# relevant file paths
+data_dir = './data'
+unprocessed_edge_list = os.path.join(data_dir, 'edgeList.csv')
+
+# map of Pa gene names to generic/not generic status, generated by Alex
+generic_gene_map = os.path.join('..', 'pseudomonas_analysis', 'annot_df.tsv')
+
+# save edge list with preprocessed weight information
+processed_edge_list = os.path.join(data_dir, 'edge_list_processed_unsigned.csv')
+
+# place to save preprocessed graph/attributes, in graph-tool binary format
+processed_graph = os.path.join(data_dir, 'eadage_generic_graph_unsigned.gt')
+
+
+# #### Load edge list and handle negative edge weights
+# 
+# Most algorithms for community detection and betweenness don't work with negative weights/correlations. Taking inspiration from WGCNA, we can solve this problem in one of two ways:
+# * Unsigned: `weight = abs(corr(g1, g2))`
+# * Signed: `weight = abs((1 + corr(g1, g2) / 2)`
+#  
+# We should probably try both eventually, but for now we'll just use the unsigned version (taking absolute value of negative edges). I tend to agree with the rationale in this BioStars post for taking absolute values (unsigned) rather than rescaling (signed approach): https://www.biostars.org/p/144078/#144088
```

Comment:
I agree. In this case I think the magnitude is what matters in terms of defining the connection

Reply:
Yeah - I don't think it will matter much in either case, since there are very few negative edges in this network, but I do think the magnitude is more important than the direction for finding modules/"pathways".